diff --git a/sites/all/modules/contrib/wysiwyg/editors/js/ckeditor-3.0.js b/sites/all/modules/contrib/wysiwyg/editors/js/ckeditor-3.0.js
index 36e0d8c..653dd9f 100644
--- a/sites/all/modules/contrib/wysiwyg/editors/js/ckeditor-3.0.js
+++ b/sites/all/modules/contrib/wysiwyg/editors/js/ckeditor-3.0.js
@@ -45,8 +45,17 @@ Drupal.wysiwyg.editor.attach.ckeditor = function(context, params, settings) {

   var $drupalToolbars = $('#toolbar, #admin-menu', Drupal.overlayChild ? window.parent.document : document);
   if (!settings.height) {
-    settings.height = $('#' + params.field).height();
+    var element = $('#' + params.field)[0];
+    if ((element.tagName.toLowerCase() == 'input') && (element.type.toLowerCase() == 'text')) {
+      settings.height = 90;
+    } else {
+      settings.height = $('#' + params.field).height();
+    }
+  }
+  if (!settings.height) {
+    settings.height = 90;
   }
+
   settings.on = {
     instanceReady: function(ev) {
       var editor = ev.editor;